src/transport/PeerMessageCounter.h

```diff
@@ -135,6 +191,64 @@ class PeerMessageCounter
         }
     }
 
+    /**
+     * @brief
+     *    With the counter verified and the packet MIC also verified by the secure key, we can trust the packet and adjust
+     *    counter states including possible Rollover for Groups communications.
+     *
+     * @pre Verify(counter) == CHIP_NO_ERROR
+     */
+    void CommitWithRollOver(uint32_t counter)
+    {
+        if (counter <= mSynced.mMaxCounter)
+        {
+            // Take into account rollover
+            uint32_t offset         = mSynced.mMaxCounter - counter;
+            uint32_t rollOverOffset = ((UINT32_MAX - mSynced.mMaxCounter) + counter);
+            if (offset < CHIP_CONFIG_MESSAGE_COUNTER_WINDOW_SIZE)
+            {
+                mSynced.mWindow.set(offset);
+            }
+            else if (rollOverOffset < CHIP_CONFIG_MESSAGE_COUNTER_WINDOW_SIZE)
+            {
+                // Roll over occured
+                mSynced.mMaxCounter = counter;
+                mSynced.mWindow <<= rollOverOffset;
+                mSynced.mWindow.set(0);
+            }
+            else
+            {
+                // Roll over occured with a leap bigger than CHIP_CONFIG_MESSAGE_COUNTER_WINDOW_SIZE
+                mSynced.mMaxCounter = counter;
+                mSynced.mWindow.reset();
+                mSynced.mWindow.set(0);
+            }
+        }
+        else
+        {
+            uint32_t offset = mSynced.mMaxCounter - counter;
+            if (offset < CHIP_CONFIG_MESSAGE_COUNTER_WINDOW_SIZE)
+            {
+                mSynced.mWindow.set(offset);
+            }
+            else
+            {
+                // Not a bit inside the window.  Since we are committing, this is a new mMaxCounter value.
```

Comment:
This codepath now works whether `counter <= mSynced.mMaxCounter` is true or not.  Why do we still have the big `if (counter <= mSynced.mMaxCounter)` block?

Reply:
Removed the block